feat: add installation, usage and start docs (#116)
mainmatter/sheepdog: docs/src/content/docs/getting-started/what-it-is.mdx

Reviewer: Very nit-picky but `what-is-it` sounds more normal here

---
title: What it is?
description: What problem does `@sheepdog/svelte` solves.
---

import Mainmatter from '@assets/Mainmatter.astro';

> `@sheepdog/svelte` is a library that aims to solve a problem that every developer will face at one
> point or another: deal with async and concurrent task.

Reviewer: ```suggestion
`@sheepdog/svelte` is a library that aims to solve a problem that every developer will face at one
point or another: handling async and concurrent tasks.
```

> `@sheepdog/svelte` supplies a simple way to introduce cancellable concurrency into your app. Not
> only we provide the "cancellability" that is missing from normal Promises, `@sheepdog/svelte` also
> provides a public API that allows you to observe the running state of your task without having to
> set a single flag manually.

Reviewer: ```suggestion
`@sheepdog/svelte` supplies a simple way to introduce cancellable concurrency into your app. Not
only do we provide the "cancellability" that is missing from normal Promises, `@sheepdog/svelte` also
provides a public API that allows you to observe the running state of your task without having to
set a single flag manually.
```

> Tasks that live on components are automatically cancelled when their context is destroyed, meaning
> you don't need to worry about the clean up - we've got you covered.

Reviewer: ```suggestion
Tasks that live on components are automatically canceled when their context is destroyed, meaning
you don't need to worry about the clean up - we've got you covered.
```
